cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/AuthBinding.java

```diff
@@ -103,11 +107,111 @@
   private final String instanceName;
   private final String sentryAdminGroup;
 
-  AuthBinding(String sentrySite, String instanceName, String sentryAdminGroup) {
+  private final LoadingCache<Principal, Set<String>> groupCache;
```

Comment:
What I meant is that the implementation can change to use LDAP lookup or something. 

Also since we are removing grants/revokes, we have to wait for the cache timeout for any change - group, role or privilege. We could reduce the timeout for cache on the master side if required, can you file a JIRA for this?